elements/lisk-chain/src/state_store/state_store.ts

```diff
@@ -12,78 +12,231 @@
  * Removal or modification of this copyright notice is prohibited.
  */
 
-import { codec } from '@liskhq/lisk-codec';
-import { BatchChain, formatInt } from '@liskhq/lisk-db';
-import { BlockHeader, StateDiff, UpdatedDiff } from '../types';
-import { AccountStore } from './account_store';
-import { ChainStateStore } from './chain_state_store';
-import { ConsensusStateStore } from './consensus_state_store';
-import { DataAccess } from '../data_access';
-import { stateDiffSchema } from '../schema';
-import { concatDBKeys } from '../utils';
-import { DB_KEY_DIFF_STATE } from '../db_keys';
-
-interface AdditionalInformation {
-	readonly lastBlockHeaders: ReadonlyArray<BlockHeader>;
-	readonly networkIdentifier: Buffer;
-	readonly lastBlockReward: bigint;
-	readonly defaultAccount: Record<string, unknown>;
+import { codec, Schema } from '@liskhq/lisk-codec';
+import { NotFoundError as DBNotFoundError } from '@liskhq/lisk-db';
+import { StateDiff } from '../types';
+import { CacheDB } from './cache_db';
+import { NotFoundError } from './errors';
+import { DatabaseReader, DatabaseWriter } from './types';
+import { copyBuffer } from './utils';
+
+// TODO: Update when other db keys are defined
+export const DB_PREFIX_STATE_STORE = Buffer.from([5]);
+
+export interface IterationOptions {
```

Comment:
I think this sounds better
```suggestion
export interface IterateOptions {
```